### PR TITLE
Fix rmtree to remove directories with read-only files

### LINKED ISSUE #5524: Race condition error while creating basetemp concurrently
I've never seen this error, and can't find a way to reproduce it:

```
10:55:07 INTERNALERROR> Traceback (most recent call last):
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\_pytest\main.py", line 203, in wrap_session
10:55:07 INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\hooks.py", line 289, in __call__
10:55:07 INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\manager.py", line 68, in _hookexec
10:55:07 INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\manager.py", line 62, in <lambda>
10:55:07 INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\callers.py", line 208, in _multicall
10:55:07 INTERNALERROR>     return outcome.get_result()
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\callers.py", line 80, in get_result
10:55:07 INTERNALERROR>     raise ex[1].with_traceback(ex[2])
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\pluggy\callers.py", line 187, in _multicall
10:55:07 INTERNALERROR>     res = hook_impl.function(*args)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\xdist\dsession.py", line 81, in pytest_sessionstart
10:55:07 INTERNALERROR>     nodes = self.nodemanager.setup_nodes(putevent=self.queue.put)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\xdist\workermanage.py", line 68, in setup_nodes
10:55:07 INTERNALERROR>     nodes.append(self.setup_node(spec, putevent))
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\xdist\workermanage.py", line 77, in setup_node
10:55:07 INTERNALERROR>     node.setup()
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\xdist\workermanage.py", line 244, in setup
10:55:07 INTERNALERROR>     basetemp = self.config._tmpdirhandler.getbasetemp()
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\_pytest\tmpdir.py", line 117, in getbasetemp
10:55:07 INTERNALERROR>     return py.path.local(self._tmppath_factory.getbasetemp().resolve())
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\_pytest\tmpdir.py", line 65, in getbasetemp
10:55:07 INTERNALERROR>     ensure_reset_dir(basetemp)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\site-packages\_pytest\pathlib.py", line 41, in ensure_reset_dir
10:55:07 INTERNALERROR>     path.mkdir()
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\pathlib.py", line 1248, in mkdir
10:55:07 INTERNALERROR>     self._accessor.mkdir(self, mode)
10:55:07 INTERNALERROR>   File "W:\Miniconda\envs\_dev_eden-win64-py36\lib\pathlib.py", line 387, in wrapped
10:55:07 INTERNALERROR>     return strfunc(str(pathobj), *args)
```

It just happen on the CI server at work. Of course we're using xdist.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #5524

### CODE REVIEW

File: src/_pytest/pathlib.py [hunk 1]
```diff
@@ -37,10 +37,15 @@ def ensure_reset_dir(path):
 
 
 def rmtree(path, force=False):
```
Comment (nicoddemus) [inline src/_pytest/pathlib.py]: @RonnyPfannschmidt I see that nobody passes `force=False` on the codebase, how about removing this parameter and always assume `force=True`?

Comment (RonnyPfannschmidt) [reply src/_pytest/pathlib.py]: Go for it

Comment (nicoddemus) [reply src/_pytest/pathlib.py]: Done

File: src/_pytest/pathlib.py [hunk 2]
```diff
@@ -32,17 +32,44 @@ def ensure_reset_dir(path):
...
+                chmod_w(parent)
+                # stop when we reach the original path passed to rm_rf
+                if parent == path:
+                    break
+
+        chmod_w(p)
+        try:
+            # retry the function that failed
+            function(str(p))
+        except Exception:
```
Comment (RonnyPfannschmidt) [inline src/_pytest/pathlib.py]: we should trigger a warning, simply to indicate that it couldnt loose the items, that would massively help the users

Comment (nicoddemus) [reply src/_pytest/pathlib.py]: Good idea, done. 👍

**Comment by nicoddemus** [comment]
Backport: https://github.com/pytest-dev/pytest/pull/5597

**Comment by blueyed** [comment]
See https://github.com/pytest-dev/pytest/issues/5626 for an issue with this.

**Pull request merged**